### ISSUE TITLE
Printer compatibility

### ISSUE THREAD

**Issue by simmogs** [escpos/escpos#1]
This looks like a brilliant gem. Could you confirm which printers you know work with this?

I've tried your sample code with my Brother TD-4000 but so far it's stubbornly silent.

**Comment by svoboda-jan** [comment]
This gem should work with any `ESC/POS` compatible printer.

Brother TD-4000 uses a different set of printer-control commands `ESC/P`, not compatible with `ESC/POS`.

**Comment by simmogs** [comment]
Thanks for the quick response. Excuse my naivety, but is it simply a case of a different set of commands in this file (https://github.com/escpos/escpos/blob/master/lib/escpos.rb), or is not that straight-forward?

**Comment by svoboda-jan** [comment]
Yes, that is true for most basic formatting commands.

As an example to underline text:

`ESC/POS`:

``` ruby
TXT_UNDERL_OFF               = [ 0x1b, 0x2d, 0x00 ]        # Underline font OFF
TXT_UNDERL_ON                = [ 0x1b, 0x2d, 0x01 ]        # Underline font 1
```

`ESC/P`:

``` ruby
# same command for ON and OFF (from linked PDF, page 7)
TXT_UNDERL_OFF               = [ 0x1b, 0x2d ]              # Apply/cancel underlining
TXT_UNDERL_ON                = [ 0x1b, 0x2d ]              # Apply/cancel underlining
```

`ESC/P` command reference can be found here:
http://download.brother.com/welcome/docp000579/cv_td4000_eng_escp_120.pdf

Looking at the `ESC/P` command reference, it seems that some printer-control commands are the same in both sets, eg. `HW_INIT`, but unfortunately I do not have access to any `ESC/P` printer, so I can't do any real tests.

**Comment by bunnakal** [comment]
Hi creator,
Currently, I have developed POS system for a coffee-shop, and have kind of receipt printing.
They have bought a small thermal printer model Xprinter XP-C230, I have looked at the device manual that stated that it compatible with ESC/POS, but when I have implement with your gem and click button print there is nothing. it still keep silently. I have go through your document I really don't know which point I have missed or any other configuration.
Note: if print text from notepad it just working find and print to printer, but when I print from my RoR application it won't work. 

I am waiting to hearing from you.
Thank you in advance.
